scheduler/src/cook/mesos.clj

```diff
@@ -319,12 +321,20 @@
                                  (or server-port server-https-port) \#
                                  (if server-port "http" "https") \#
                                  (java.util.UUID/randomUUID)))
-    (.autoRequeue leader-selector)
-    (.start leader-selector)
-    (log/info "Started the leader selector")
+    (if api-only?
+      ; If this is an api-only instance, we still want all the leader election configuration to be set,
+      ; without actually joing the election process.
+      ; This allows the instance to correctly forward client requests to the leader as necessary.
+      (log/info "Will not join leader election, running as api-only")
+      (do
+        (.autoRequeue leader-selector)
+        (.start leader-selector)
+        (log/info "Started the leader selector")
+        ))
```

Comment:
reformat to bring up to the previous line

Reply:
Yikes, not sure how that happened. I fixed it.